src/android/CameraLauncher.java

```diff
@@ -178,7 +181,7 @@ else if ((this.srcType == PHOTOLIBRARY) || (this.srcType == SAVEDPHOTOALBUM)) {
                     // preserve the original exif data and filename in the modified file that is
                     // created
                     if(this.mediaType == PICTURE && (this.destType == FILE_URI || this.destType == NATIVE_URI)
-                            && fileWillBeModified() && !cordova.hasPermission(permissions[0])) {
+                            && fileWillBeModified() && !cordova.hasPermission(permissions[SAVE_TO_ALBUM_SEC])) {
```

Comment:
I don't think we should be using SAVE_TO_ALBUM_SEC and TAKE_PICTURE_SEC for the indices in the permissions array since those values mean something different. The action after the permissions is not determined by the permissions requested (both the camera and the photolibrary sources require READ permission, for example). Could you define some different constants at the top of the file; maybe CAMERA_PERMISSION_INDEX and READ_PERMISSION_INDEX?